Language/C/Quote/Base.hs

```diff
@@ -224,19 +224,30 @@ qqIdE _                 = Nothing
 qqDeclSpecE :: C.DeclSpec -> Maybe (Q Exp)
 qqDeclSpecE (C.AntiDeclSpec v _) = Just $ antiVarE v
 qqDeclSpecE (C.AntiTypeDeclSpec extraStorage extraTypeQuals v _) =
-    Just [|let C.Type (C.DeclSpec storage typeQuals typeSpec loc) _ _
-                   = $(antiVarE v)
-           in
-             C.DeclSpec (storage ++ $(dataToExpQ qqExp extraStorage))
-                        (typeQuals ++ $(dataToExpQ qqExp extraTypeQuals))
-                        typeSpec
-                        loc
+    Just [|
+        case $(antiVarE v) of
+            C.Type (C.DeclSpec storage typeQuals typeSpec loc) _ _ ->
+                 C.DeclSpec
+                     (storage ++ $(dataToExpQ qqExp extraStorage))
+                     (typeQuals ++ $(dataToExpQ qqExp extraTypeQuals))
+                     typeSpec
+                     loc
+
+            x -> error
+                   $ "Impossible happened, expected C.Type but got "
```

Comment:
This one should maybe say expected `C.Type (C.DeclSpec {})`. Can improve iff needed / we don't figure out neater solution.

Reply:
I'm OK with this.  It's an internal error, isn't it?

Reply:
It is and very unlikely to ever pop up I think. But I've made it nicer!